packages/@react-spectrum/table/src/TableViewBase.tsx

```diff
@@ -1128,9 +1128,10 @@ function TableRow({item, children, layoutInfo, parent, ...otherProps}) {
     shouldSelectOnPressUp: isTableDraggable
   }, state, ref);
 
-  let isDisabled = !allowsSelection && !hasAction;
+  let isDisabled = state.selectionManager.isDisabled(item.key);
+  let isInteractive = !isDisabled && (hasAction || allowsSelection || isTableDraggable);
```

Comment:
Note that these changes mean that there are some changes to Table behavior from before. For instance, we used to block dragging a row if it was disabled via `disabledKeys`. However, this is inconsistent from ListView and doesn't really make sense since by default `disabledBehavior="selection"` so `disabledKeys` should only really be affecting selectability.